libs/locale.cmake

```diff
@@ -59,6 +59,7 @@ _add_boost_lib(
 # Convenience interface library to link deps to both main library and tests
 add_library(Boost_locale_deps INTERFACE)
 target_link_libraries(Boost_locale PRIVATE Boost_locale_deps)
+install(TARGETS Boost_locale_deps DESTINATION lib EXPORT boost-libs)
```

Comment:
Per one of my last comments, as I was considering implementing config packages for your projects myself, the way I would have done it is by adding a custom `_install_boost_component()` function that does something like:

```
function( _install_boost_component component )
    set( export_name Boost${component}Targets )
    install( TARGETS Boost_${component} EXPORT ${export_name}
        RUNTIME DESTINATION bin
        LIBRARY DESTINATION lib
        ARCHIVE DESTINATION lib
        INCLUDES DESTINATION include
    )

    install(EXPORT ${export_name} DESTINATION share/boost-${BOOST_VERSION}/cmake )

    # TODO:
    # install(FILES) for the header files since `INCLUDES DESTINATION` just maps include path and doesn't actually install any files
endfunction()
```

I think this keeps things more modular, and allows you to use the `COMPONENTS` section of `find_package()` more reliably (i.e. `find_dependency()` macro)